scaaml/capture/ascad/ascad_capture.py

```diff
@@ -0,0 +1,274 @@
+                        dataset=dataset)
+                    capture_runner.capture()
+
+
+def capture_ascad_default_parameters(
+        firmware_sha256: str,
+        crypto_implementation=AESSBOX,
+        algorithm: str = 'simpleserial-aes',
+        version: int = 1,
+        root_path: str = '/mnt/storage/chipwhisperer',
+        url: str = '',
+        firmware_url: str = '',
+        paper_url: str = '',
+        licence: str = "https://creativecommons.org/licenses/by/4.0/",
+        examples_per_shard: int = 64,
+        measurements_info=None,
+        repetitions: int = 1,
+        chip_id: int = 620510,  # STM32F415 Training
+        train_keys: int = 4 * 1024,
+        train_plaintexts: int = 256,
+        holdout_keys: int = 0 * 1024,
+        holdout_plaintexts: int = 1) -> None:
+    """SCAAML STM32F415 ASCAD specific defaults for capture_ascad_dataset."""
+    architecture: str = 'CW308_STM32F415'
+    implementation: str = 'ASCAD'
+    shortname: str = 'stm32f415_hwaes'
+    description: str = 'SCAAML HWAES'
+
+    # Parameters for the scope
+    samples: int = 80_000
+    offset: int = 0
+    sample_rate: float = 1_000_000_000.0
+    trace_probe_range: float = 0.05
+    trigger_range: float = 5.0
+    trigger_level: float = 1.9
+    capture_info = {
+        'samples': samples,
+        'trigger_level': trigger_level,
+        'trigger_range': trigger_range,
+        'sample_rate': sample_rate,
+        'offset': offset,
+        'trace_probe_range': trace_probe_range,
+    }
+
+    capture_ascad_dataset(firmware_sha256=firmware_sha256,
+                          architecture=architecture,
+                          implementation=implementation,
+                          shortname=shortname,
+                          description=description,
+                          crypto_implementation=crypto_implementation,
+                          algorithm=algorithm,
+                          version=version,
+                          root_path=root_path,
+                          url=url,
+                          firmware_url=firmware_url,
+                          paper_url=paper_url,
+                          licence=licence,
+                          examples_per_shard=examples_per_shard,
+                          measurements_info=measurements_info,
+                          capture_info=capture_info,
+                          repetitions=repetitions,
+                          chip_id=chip_id,
+                          train_keys=train_keys,
+                          train_plaintexts=train_plaintexts,
+                          holdout_keys=holdout_keys,
+                          holdout_plaintexts=holdout_plaintexts)
+
+
+def load_firmware():
+    """Load firmware."""
+    FIRMWARE = "chipwhisperer-firmware/simpleserial-aes/simpleserial-aes_CW308-STM32F4_HWAES.hex"
+    cwscope = cw.scope()
+    cwscope.default_setup()
+    try:
+        cw.program_target(cwscope, cw.programmers.STM32FProgrammer, FIRMWARE)
```

Comment:
Loading firmware through the Chipwhisperer has been quite unreliable in the past, not to mention that it only supports a handful of targets.
Consider moving to the code I wrote to program the chips using an external programmer. It should support all the target chips that can be purchased with the Chipwhisperer and provide a consistant API and Notebook widgets :)